### ISSUE TITLE
Integrate with AngularJS - directive

### ISSUE THREAD

**Issue by lukekrupa** [tpaksu/jquery.slidey#1]
Hello,

Bootstrap template I'm currently using uses jquery.slidey as one of sliders. 

Problem occured when I was trying to use this slider in angular's partial view.

So to do that, I wrote angular's directive in controller `home.controller.js`:

```javascript
app.directive('slidey', function () {
   return {
      restrict: 'A',
      link: function (scope, element, attrs) {
         element.slidey({
            interval: 8000,
            listCount: 5,
            showList: true
         });
      }
   }
})
```

and in my partial view `home.html` I've added another piece of code:
```html
<div slidey id="slidey" style="display:none;">
                    <ul>
                        <li><img src="assets/1.jpg"><p class='title'>Star Wars</p><p class='description'>After rescuing Han Solo from the palace of Jabba the Hutt, the rebels attempt to destroy the second Death Star, while Luke struggles to make Vader return from the dark side of the Force.</p></li>
                        <li><img src="assets/2.jpg"><p class='title'>Silicon Valley</p><p class='description'>Mike Judge brings his irreverent brand of humor to HBO in the new comedy series SILICON VALLEY. Partially inspired by Judge's own experiences as a Silicon Valley engineer in the late '80s, the show is a collaboration between Judge and Alec Berg.</p></li>
                        <li><img src="assets/3.jpg"><p class='title'>Fringe</p><p class='description'>The Fringe team explore the blurring line between the possible and the impossible. They are encountering glitches in the world they are living in, face threats from a parallel universe and try to prevent events that will lead to the destruction of their own universe.</p></li>
                        <li><img src="assets/4.jpg"><p class='title'>Assassin's Creed</p><p class='description'>When Callum Lynch (Michael Fassbender) explores the memories of his ancestor Aguilar and gains the skills of a Master Assassin, he discovers he is a descendant of the secret Assassins society.</p></li>
                        <li><img src="assets/5.jpg"><p class='title'>Star Wars</p><p class='description'>Three decades after the defeat of the Galactic Empire, a new threat arises. The First Order attempts to rule the galaxy and only a ragtag group of heroes can stop them, along with the help of the Resistance.</p></li>
                        <li><img src="assets/6.jpg"><p class='title'>Continuum</p><p class='description'>Continuum is a one-hour police drama centered on Kiera Cameron, a regular cop from 65 years in the future who finds herself trapped in present day Vancouver. She is alone, a stranger in a strange land, and has eight of the most ruthless criminals from the future, known as Liber8, loose in the city.

Lucky for Kiera, through the use of her CMR (cellular memory recall), a futuristic liquid chip technology implanted in her brain, she connects with Alec Sadler, a seventeen-year-old tech genius. When Kiera calls and Alec answers, a very unique partnership begins.

Kiera's first desire is to get "home." But until she figures out a way to do that, she must survive in our time period and use all the resources available to her to track and capture the terrorists before they alter history enough to change the course of the future. After all, what's the point of going back if the future isn't the one you left?</p></li>
                        <li><img src="assets/7.jpg"><p class='title'>Agents of S.H.I.E.L.D.</p><p class='description'>Clark Gregg reprises his role of Agent Phil Coulson from Marvel's feature films, as he assembles a small, highly select group of Agents from the worldwide law-enforcement organization known as S.H.I.E.L.D. Together they investigate the new, the strange, and the unknown across the globe, protecting the ordinary from the extraordinary. Coulson's team consists of Agent Grant Ward, highly trained in combat and espionage Agent Melinda May expert pilot and martial artist; Agent Leo Fitz, brilliant engineer; and Agent Jemma Simmons, genius bio-chemist. Joining them on their journey into mystery is new recruit and computer hacker, Skye.</p></li>
                        <li><img src="assets/8.jpg"><p class='title'>Falling Skies</p><p class='description'>The series tells the story of the aftermath of a global invasion by several races of extraterrestrials that neutralizes the world's power grid and technology, quickly destroys the combined militaries of all the world's countries, and apparently kills over 90% of the human population within a few days.</p></li>
                        <li><img src="assets/9.jpg"><p class='title'>Avengers</p><p class='description'>Now that there is the Tec-9, a crappy spray gun from South Miami. This gun is advertised as the most popular gun in American crime. Do you believe that shit? It actually says that in the little book that comes with it: the most popular gun in American crime. Like they're actually proud of that shit. </p></li>
                        <li><img src="assets/10.jpg"><p class='title'>Friends</p><p class='description'>Six young people from New York, on their own and struggling to survive in the real world, find the companionship, comfort and support they get from each other to be the perfect antidote to the pressures of life.</p></li>
                    </ul>                    
                </div>
```

Of course I also added to main view `index.html` proper includes:
```html
    <script src="dist/jquery.min.js"></script>
    <!-- jQuery Slidey 0.0.1 -->
    <link href="dist/jquery.slidey.min.css" rel="stylesheet">
    <link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css" rel="stylesheet">
    <!--<script src="dist/jquery.slidey.min.js"></script>-->
    <script src="jquery.slidey.js"></script>
    <script src="dist/jquery.dotdotdot.min.js"></script>
```

But at the end of the day, slider is not showing.

Have you got any ideas how to fix it?

Thanks, 
Luke

**Comment by lukekrupa** [comment]
Okay, I found the cause of mentioned problem. 

jquery.slidey is using `size()` method (in `fetchSlides` function). According to https://api.jquery.com/size/ this method has been removed since jQuery 3.0, which is exactly the same version I'm using in my project.

I think it is a good idea to introduce compatibility with jQuery 3.x. Besides, `.length` property is available in jQuery from very beginning, so I don't see any reason why it shouldnt be compatible with 3.
